Common/Data/Custom/Intrinio/IntrinioEconomicData.cs

```diff
@@ -0,0 +1,218 @@
+        private readonly string _password = Config.Get("intrinio-password");
+        private readonly string _user = Config.Get("intrinio-username");
+
+        private readonly string _baseUrl = @"https://api.intrinio.com/historical_data.csv?";
+
+        private bool _firstTime = true;
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="IntrinioEconomicData" /> class.
+        /// </summary>
+        public IntrinioEconomicData() : this(IntrinioDataTransformation.Level)
+        {
+        }
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="IntrinioEconomicData" /> class.
+        /// </summary>
+        /// <param name="dataTransformation">The item.</param>
+        public IntrinioEconomicData(IntrinioDataTransformation dataTransformation)
+        {
+            _dataTransformation = dataTransformation;
+        }
+
+
+        /// <summary>
+        ///     Return the URL string source of the file. This will be converted to a stream
+        /// </summary>
+        /// <param name="config">Configuration object</param>
+        /// <param name="date">Date of this source file</param>
+        /// <param name="isLiveMode">true if we're in live mode, false for backtesting mode</param>
+        /// <returns>
+        ///     String URL of source file.
+        /// </returns>
+        public override SubscriptionDataSource GetSource(SubscriptionDataConfig config, DateTime date, bool isLiveMode)
+        {
+            SubscriptionDataSource subscription;
+            if (_firstTime)
+            {
+                var order = string.Empty;
+                if (isLiveMode)
+                {
+                    order = "desc";
+                }
+                else
+                {
+                    order = "asc";
+                    _firstTime = false;
+                }
+
+                var item = GetStringForDataTransformation(_dataTransformation);
+                var url = string.Format("{0}identifier={1}&item={2}&sort_order={3}", _baseUrl,
+                                        config.Symbol.Value, item, order);
+                var byteKey = Encoding.ASCII.GetBytes(string.Format("{0}:{1}", _user, _password));
+                var authorizationHeaders = new List<KeyValuePair<string, string>>
+                {
+                    new KeyValuePair<string, string>("Authorization",
+                                                     string.Format("Basic ({0})", Convert.ToBase64String(byteKey)))
+                };
+
+                subscription = new SubscriptionDataSource(url, SubscriptionTransportMedium.RemoteFile, FileFormat.Csv,
+                                                          authorizationHeaders);
+                Thread.Sleep(millisecondsTimeout: 1000);
+            }
+            else
+            {
+                subscription = new SubscriptionDataSource("", SubscriptionTransportMedium.RemoteFile);
```

Comment:
We might want a comment here about what we're doing... and if it *works* perhaps returning null is more descriptive of why we're doing this. IIRC,this is so that we do not hit the intrinio API again

Reply:
Is it returns null throws error:
`ParallelRunnerWorker.ThreadEntry():  System.NullReferenceException: Object reference not set to an instance of an object.`